azurerm/internal/services/apimanagement/resource_arm_api_management.go

```diff
@@ -514,6 +537,12 @@ func resourceArmApiManagementServiceCreateUpdate(d *schema.ResourceData, meta in
 			CustomProperties:       customProperties,
 			Certificates:           certificates,
 			HostnameConfigurations: hostnameConfigurations,
+			VirtualNetworkType:     virtualNetworkTypeStringToType(utils.String(d.Get("vnet_type").(string))),
```

Comment:
We should be able to go
```suggestion
			VirtualNetworkType:     apimanagement.VirtualNetworkType(d.Get("vnet_type").(string))),
```